gabbi/case.py

```diff
@@ -225,29 +245,33 @@ def _environ_replace(self, message, escape_regex=False):
                        self._regex_replacer(self._environ_replacer,
                                             escape_regex),
                        message)
-        try:
-            if '.' in value:
-                value = float(value)
-            else:
-                value = int(value)
+        if self.cast:
+            return self._cast_value(value, message)
+        else:
+            try:
+                if '.' in value:
+                    value = float(value)
+                else:
+                    value = int(value)
+                return value
+            except ValueError:
+                pass
+            if value.lower() == "false":
+                return False
+            if value.lower() == "true":
+                return True
+            if value.lower() == "null":
+                return None
```

Comment:
minor nit:
```python
return {
    "true": True,
    "false": False,
    "null": None
}.get(value.lower(), value)
```
seems more readable/idiomatic?

Reply:
Perhaps, but code was already there in the previous way and changing it now would be an unrelated change.

Also that looks icky.